.github/workflows/docker-image.yml

```
@@ -0,0 +1,35 @@
+name: Publish Docker image
+
+on:
+  push:
+    branches:
+      - develop
```

This is correct, but if you want to test this workflow before merging then you should use this:
```suggestion
on:
  pull_request:
    types: [ synchronize ]
```

Just make sure to switch it back when you're done.